lib/utils/errors.js

```diff
@@ -35,6 +63,55 @@ class ExceededMaxFileSizeError extends EvervaultError {}
 
 class TokenCreationError extends EvervaultError {}
 
+const mapFunctionFailureResponseToError = ({ error, id }) => {
+  if (error) {
+    if (error.message.includes('The function failed to initialize.')) {
+      throw new FunctionInitializationError(error.message, error.stack, id);
+    }
+    throw new FunctionRuntimeError(error.message, error.stack, id);
+  }
+  throw new EvervaultError('An unknown error occurred.');
+};
+
+const mapFunctionResponseToError = ({ code, detail }) => {
+  if (code === 'unauthorized') {
+    throw new UnauthorizedError(detail);
+  }
+  if (code === 'forbidden') {
+    if (detail == 'Run requested by an IP address which is not whitelisted.') {
```

Comment:
We shouldn't use the detail field to identify the error, this is what the "code" field should be used for. I think this should be a namespaced error under function instead of being a `forbidden` error.

Reply:
I thought this too! But I wasn't sure how that fit into the api spec because we currently have one `code` per `statusCode`? Agree it's the right approach, just wasn't sure how to fit it into the spec

Reply:
We're actually revamping the specs slightly to support several error codes per status code (this was initially just a limitation caused by the way we were structuring examples). It's totally acceptable to have several errors per status code! :)